node/validator_store.go

```diff
@@ -0,0 +1,540 @@
+			LevelDBSyncWrites:             config.LevelDBSyncWritesV2,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to create new tablestore: %w", err)
+		}
+	}
+
+	// Set up migration if necessary.
+	var migrationComplete time.Time
+	if config.LittDBEnabled && levelDBExists {
+		// Both DBs are in play, meaning we are either about to start a migration or already in the middle of one.
+
+		migrationKeyBuilder, err := levelDB.GetKeyBuilder(MigrationTableName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get key builder for migration: %v", err)
+		}
+		migrationKey := migrationKeyBuilder.Key([]byte("migrationCompleteTime"))
+
+		if !littDBExists {
+			// This is the first time we are starting up with littDB enabled and there is potentially data still
+			// in the levelDB. Start the data migration from levelDB to littDB.
+
+			migrationComplete = timeSource().Add(ttl)
+			logger.Infof("Begining data migration from levelDB to littDB. Migration will be completed at %s",
+				migrationComplete)
+
+			migrationCompleteUnix := migrationComplete.Unix()
+			err = levelDB.Put(migrationKey, []byte(fmt.Sprintf("%d", migrationCompleteUnix)))
+			if err != nil {
+				return nil, fmt.Errorf("failed to put migration key: %v", err)
+			}
+
+		} else {
+			// A data migration from levelDB to littDB is currently in progress.
+
+			migrationCompleteString, err := levelDB.Get(migrationKey)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get migration complete time: %v", err)
+			}
+			migrationCompleteUnix, err := strconv.ParseUint(string(migrationCompleteString), 10, 64)
+			if err != nil {
+				return nil, fmt.Errorf("failed to read migration complete time: %v", err)
+			}
+			migrationComplete = time.Unix(int64(migrationCompleteUnix), 0)
+
+			logger.Infof(
+				"Data migration from levelDB to littDB is in progress. Migration will be completed at %s",
+				migrationComplete)
+		}
+	}
+
+	// Start littDB.
+	// The ordering of this step is important. By starting littDB, we will create the littDB directory, which
+	// will cause the littDBExists variable to be true for all future runs. It's important to have written the
+	// migration complete timestamp prior to this happening so that a crash during startup does not leave the
+	// migration in a broken state.
+	if config.LittDBEnabled {
+		logger.Infof("Using littDB at %s", littDBPath)
+
+		littConfig, err := litt.DefaultConfig(littDBPath)
+		littConfig.ShardingFactor = 1
+		littConfig.MetricsEnabled = true
+		littConfig.MetricsRegistry = registry
+		littConfig.MetricsNamespace = "node_littdb"
+		littConfig.Logger = logger
+		littConfig.DoubleWriteProtection = config.LittDBDoubleWriteProtection
+		if err != nil {
+			return nil, fmt.Errorf("failed to create new litt config: %w", err)
+		}
+
+		littDB, err = littbuilder.NewDB(littConfig)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create new litt store: %w", err)
+		}
+
+		chunkTable, err = littDB.GetTable("chunks")
```

Comment:
Any reason you hardcode "chunks" here, instead of having a const defined above, like for the other table names?

Reply:
No reason, moved magic strings into constants.